exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/MapVector.java

```diff
@@ -355,17 +353,19 @@ public void generateTestData(int values) { }
 
   @Override
   public void clear() {
-    valueCount = 0;
-    for (ValueVector v : getChildren()) {
+    for (final ValueVector v : getChildren()) {
```

Comment:
the logic in clear() seems duplicating close(). we should directly call close perhaps.